runtime/lua/vim/_init_packages.lua

```diff
@@ -69,6 +69,14 @@ setmetatable(vim, {
         t[key] = val
         return t[key]
       end
+    elseif key == 'version' then
+      local version = require('vim.version')
+      t[key] = setmetatable({ cmp = version.cmp, parse = version.parse }, {
+        __call = function()
+          return vim.fn.api_info().version
+        end,
+      })
```

Comment:
We could use `setmetatable(M, { stuff })` inside `runtime/lua/vim/version.lua`, instead of hiding it here.